tests/integration/rb_client.py

```diff
@@ -0,0 +1,151 @@
+                             ]
+    }
+}
+
+scheduler_single_gpu = {
+    "bloom-560m": {
+        "input_texts": "When your legs don'\''t work like they used to before And I can'\''t sweep you off",
+        "parameters": {"max_new_tokens": 40, "decoding_strategy": "contrastive"},
+        "expected_word_count": 20,
+        "concurrent_clients": 4
+    }
+}
+
+scheduler_multi_gpu = {
+    "gpt-j-6b": {
+        "input_texts": "When your legs don't work like they used to before And I can't sweep you off",
+        "parameters": {"max_new_tokens": 40, "decoding_strategy": "contrastive"},
+        "expected_word_count": 20,
+        "concurrent_clients": 4,
+    }
+}
+
+
+def send_json(data, output_file, concurrent_clients=1):
+    endpoint = f"http://127.0.0.1:8080/invocations"
+
+    commands = ["./awscurl",
+                "-X",
+                "POST",
+                endpoint,
+                "-H content-type : application/json",
+                "-d",
+                str(json.dumps(data)),
+                "-c",
+                str(concurrent_clients),
+                "-o",
+                output_file]
+    process = sp.Popen(commands)
+    return process
+
+
+def compare_output_in_file(filename, expected_output):
+    with open(filename, 'r') as file:
+        actual_output = file.read()
+        if actual_output.startswith(expected_output):
+            return True
+        else:
+            return False
+
+
+def count_words_in_file(filename):
+    with open(filename, 'r') as file:
+        content = file.read()
+        word_count = len(content.split())
+    return word_count
+
+
+# sends the multiple requests with same data json concurrently
+def test_concurrent_with_same_reqs(model, test_spec, spec_name):
+    if model not in test_spec:
+        raise ValueError(
+            f"{args.model} is not one of the supporting models {list(test_spec.keys())}"
+        )
+
+    spec = test_spec[model]
+    data = {"inputs": spec["input_texts"], "parameters": spec.get("parameters", {})}
+    output_file = f"outputs/{model}-{spec_name}-output.txt"
+
+    process = send_json(data=data, output_file=output_file, concurrent_clients=spec.get("concurrent_clients"))
+    process.wait()
+    if process.returncode != 0:
+        raise Exception("Prediction request failed.")
+
+    if "expected_word_count" in spec:
+        expected_word_count = spec["expected_word_count"]
+        if count_words_in_file(output_file) < expected_word_count:
+            raise AssertionError("Did not produce the expected number of words")
+
+
+# sends multiple requests with different data json concurrently
+def test_concurrent_with_mul_reqs(model, test_spec, spec_name):
+    if model not in test_spec:
+        raise ValueError(
+            f"{args.model} is not one of the supporting models {list(test_spec.keys())}"
+        )
+
+    processes = []
+    spec = test_spec[model]
+    for index in range(len(spec.get("input_texts"))):
```

Comment:
why multiple processes and concurrent clients used at the same time?

Reply:
Multiple process with one concurrent client. Need to send different requests with different decoding strategy requests. For example, greedy and contrastive can be sent to the SeqBatchScheduler at the same time. But SeqBatchScheduler will call the forward separately for greedy and contrastive.  This is a scheduler only use case.